File: solutions/python/isbn-verifier/isbn_verifier.py
def is_valid(isbn: str) -> bool:
    cleaned_isbn = isbn.replace('-', '')

    if len(cleaned_isbn) != 10 or not cleaned_isbn[:-1].isdigit() or not cleaned_isbn[-1] in '0123456789X':
        return False

    digits = cleaned_isbn[:-1]
    check_char = cleaned_isbn[-1]
    check_digit = 10 if check_char == 'X' else int(check_char)

    check_sum = sum(
        int(digit) * weight
        for digit, weight in zip(digits, range(10, 1, -1))
    )
    return (check_sum + check_digit) % 11 == 0

File: solutions/go/isbn-verifier/isbn_verifier.go
package isbn

import (
	"unicode"
)

func IsValidISBN(isbn string) bool {
	number := 0
	position := 10
	for _, r := range isbn {
		switch {
		case position == 0:
			return false
		case r == '-':
			continue
		case unicode.IsDigit(r):
			digit := int(r - '0')
			number += digit * position
			position--
		case r == 'X' && position == 1:
			number += 10
			position--
		default:
			return false
		}
	}
	return position == 0 && number%11 == 0
}
